### ISSUE TITLE
recording track: missing the end of the track

### ISSUE THREAD

**Issue by oluc** [osmandapp/OsmAnd#13390]
### Description
when recording a track, at arrival, the track is displayed entirely. However after pressing "Finish", the track is missing the last part (sometimes the last 100m, sometimes the last 1km, ...)

This is not systematic, although it happened multiple times lately.

It seems similar to #12226, but in my case the track does not seem to be recorded entirely in the GPX file — the end is really missing.
#12226 referes to #12046 about slow storage, so I want to mention that my phone is a bit overlaoded and the SD card is not the fastest one, the phone became sluggish lately.

### Your Environment
OsmAnd Version: 4.0.9
Android/iOS version: 8.0.0
Device model: Samsung Galaxy A3 (2017)

**Comment by vshcherb** [comment]
https://github.com/osmandapp/OsmAnd/issues/11276

**Comment by oluc** [comment]
Is there a way we can help tackling this ?


To add a bit more information

1. I never noticed this with osmand 3.x series, it seems it came with 4.x  (and I had no Android update in the mean time)

2. what I observed seems a bit different than what's reported in the other reports: the track is truncated in the GPX file. Opening the GPX file with another app, the track is truncated too. Although the file does include the closing tags (...</trk></gpx>)

3. before I press "Finish", the recorded track is displayed on screen, so the data is in memory (it's not a GPS failure), it's only after pressing "Finish" that part of the track disappears.

The last points makes me think the issue happens when writting the GPX file.

**Comment by vshcherb** [comment]
Probably you can click share before you save the track and validate if it's same complete or not complete.

**Comment by oluc** [comment]
I did "share / save as" on the curently recording track.

In the meantime I also updated my SD card to a faster one.

What happens now is as in #12226 :
- right after "Finish" recording, **_sometimes_**, the end of the track is missing (not displayed), 
- after a restart, the end of the track is back here. (= no data loss)


About the earlier tracks (prior upgrading my SD card), the data is really missing (= data lost).
This could be due to the slow SD card. Does it make sens?

Any way, something weird happens when saving a recorded track. Maybe there should be some internal checks?

(about this specific report, feel free to keep open or just close)

**Comment by oluc** [comment]
I updated to 4.1.11 
so far, I didn't notic the issues 
(nor this one, nor #12226)

**Comment by miklcct** [comment]
Reproduced in Moto g84 5G with the latest version from Play Store

**Comment by Zirochkabila** [comment]
@miklcct try [nightly](https://download.osmand.net/latest-night-build/OsmAnd-default.apk) please. We are waiting for your feedback

**Comment by miklcct** [comment]
> [@miklcct](https://github.com/miklcct) try [nightly](https://download.osmand.net/latest-night-build/OsmAnd-default.apk) please. We are waiting for your feedback

It may be a bit difficult for me to do that on my current phone because uninstalling the play store version and installing an .apk will cause all settings to be lost.

**Comment by Zirochkabila** [comment]
@miklcct Beforehand, you can [export](https://osmand.net/docs/user/personal/import-export/) your data or use the features of the Pro subscription and transfer your data to the [cloud](https://osmand.net/docs/user/personal/osmand-cloud/)

**Comment by stragu** [comment]
I have observed this several times in the past.

Today's occurrence: the last ~1.3 km missing from a ~6.6 km track. Using 3 second intervals for recording. I could see the correct length and track before clicking "Finish". A restart did not bring back the missing data.

Using version OsmAnd~ 4.9.10 from F-Droid, on a Galaxy A20e on Android 11 (no Android updates available for this phone).

**Comment by pebogufi** [comment]
I also have this problem when cycling. 
When arriving, length of track looks ok, line of track is visible until destination. 

After saving the last part is lost, today more than 14 km, length count of km is cut, track line is cut.
I always track also with OsmTracker, so I have always complete tracks to compare.

OsmAnd~  5.1.0#5230m.apk 2025-05-15
Device Samsung S10
Android 12

**Comment by stragu** [comment]
Maintainers: maybe worth closing this as duplicate of issue #11276 and continuing the discussion there?

**Comment by pebogufi** [comment]
I made the todays cycling tour with OsmAnd~  5.1.0#5176m.apk, 2026-05-13.
Eberything fine, nothing lost.

**Comment by stragu** [comment]
I'm not so sure it's not the same underlying problem.

In my testing, I've found that:

* It happens inconsistently. Sometimes a chunk is missing, sometimes not.
* In https://github.com/osmandapp/OsmAnd/issues/11276, I reported that the missing chunk wasn't recovered after a restart, as I tried with one track. But I tried again more recently and I was able to recover the missing part. So again, another inconsistency there, possible in the sequence of events.
* The portion that is missing is variable. In my last test, it went from 10.8 to 1.7 km.

Now I am wondering if renaming the track right after finishing it is a factor. I remember the issue only happening when I rename the track directly after. Will continue to test.

**Comment by pebogufi** [comment]
I also always rename the file directly when saving.
Next days I will separate that, first save, then rename in a second step.

**Comment by pebogufi** [comment]
As @stragu reported:
> Now I am wondering if renaming the track right after finishing it is a factor. I remember the issue only happening when I rename the track directly after. Will continue to test. 

 After more tests I can confirm that:
3 times renaming directly during save - each time lost part at the end
7 times first save, then rename in a second step. - not a single loose of data.
